// @shopify/remix-oxygen/dist/production/index.cjs
'use strict';

var serverRuntime = require('@remix-run/server-runtime');

var c=new TextEncoder,d=async(e,t)=>{let r=await g(t,["sign"]),n=c.encode(e),s=await crypto.subtle.sign("HMAC",r,n),o=btoa(String.fromCharCode(...new Uint8Array(s))).replace(/=+$/,"");return e+"."+o},l=async(e,t)=>{let r=e.lastIndexOf("."),n=e.slice(0,r),s=e.slice(r+1),o=await g(t,["verify"]),a=c.encode(n),i=m(atob(s));return await crypto.subtle.verify("HMAC",o,i,a)?n:!1};async function g(e,t){return await crypto.subtle.importKey("raw",c.encode(e),{name:"HMAC",hash:"SHA-256"},!1,t)}function m(e){let t=new Uint8Array(e.length);for(let r=0;r<e.length;r++)t[r]=e.charCodeAt(r);return t}var u=serverRuntime.createCookieFactory({sign:d,unsign:l}),h=serverRuntime.createCookieSessionStorageFactory(u),y=serverRuntime.createSessionStorageFactory(u),F=serverRuntime.createMemorySessionStorageFactory(y);var A=Error.prototype.toString;Error.prototype.toString=function(){return this.stack||A.call(this)};function E({build:e,mode:t,poweredByHeader:r=!0,getLoadContext:n}){let s=serverRuntime.createRequestHandler(e,t);return async o=>{let a=o.method;if((a==="GET"||a==="HEAD")&&o.body)return new Response(`${a} requests cannot have a body`,{status:400});let i=n?await n(o):void 0,p=await s(o,i);return r&&p.headers.append("powered-by","Shopify, Hydrogen"),p}}function v(e){let t=e.headers;return {requestGroupId:t.get("request-id"),buyerIp:t.get("oxygen-buyer-ip"),cookie:t.get("cookie"),purpose:t.get("purpose")}}

Object.defineProperty(exports, "MaxPartSizeExceededError", {
	enumerable: true,
	get: function () { return serverRuntime.MaxPartSizeExceededError; }
});
Object.defineProperty(exports, "createSession", {
	enumerable: true,
	get: function () { return serverRuntime.createSession; }
});
Object.defineProperty(exports, "defer", {
	enumerable: true,
	get: function () { return serverRuntime.defer; }
});
Object.defineProperty(exports, "isCookie", {
	enumerable: true,
	get: function () { return serverRuntime.isCookie; }
});
Object.defineProperty(exports, "isSession", {
	enumerable: true,
	get: function () { return serverRuntime.isSession; }
});
Object.defineProperty(exports, "json", {
	enumerable: true,
	get: function () { return serverRuntime.json; }
});
Object.defineProperty(exports, "redirect", {
	enumerable: true,
	get: function () { return serverRuntime.redirect; }
});
Object.defineProperty(exports, "redirectDocument", {
	enumerable: true,
	get: function () { return serverRuntime.redirectDocument; }
});
Object.defineProperty(exports, "unstable_composeUploadHandlers", {
	enumerable: true,
	get: function () { return serverRuntime.unstable_composeUploadHandlers; }
});
Object.defineProperty(exports, "unstable_createMemoryUploadHandler", {
	enumerable: true,
	get: function () { return serverRuntime.unstable_createMemoryUploadHandler; }
});
Object.defineProperty(exports, "unstable_defineAction", {
	enumerable: true,
	get: function () { return serverRuntime.unstable_defineAction; }
});
Object.defineProperty(exports, "unstable_defineLoader", {
	enumerable: true,
	get: function () { return serverRuntime.unstable_defineLoader; }
});
Object.defineProperty(exports, "unstable_parseMultipartFormData", {
	enumerable: true,
	get: function () { return serverRuntime.unstable_parseMultipartFormData; }
});
exports.createCookie = u;
exports.createCookieSessionStorage = h;
exports.createMemorySessionStorage = F;
exports.createRequestHandler = E;
exports.createSessionStorage = y;
exports.getStorefrontHeaders = v;
